### PR TITLE
 Restrict unicode names #267

### PR DESCRIPTION
**Pull request by geoc0ld**
# I have made things!

## Checklist

- [ ] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [ ] I have added my changes to the `CHANGELOG.md`

### CODE REVIEW

**Comment by geoc0ld** [comment]
@sobolevn yes, of course. i will be await.

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ to the project during `#hactoberfest`. List of awesome people:
 - Forbid opening parenthesis from following keyword without space in between them
 - Forbid the use of more than 2 `for` loops within a comprehension
 - Restrict the maximum number of base classes aka mixins
+- Restrict unicode names
```
Comment (sobolevn) [inline CHANGELOG.md]: Can you please move this up to the latest version?

File: wemake_python_styleguide/violations/naming.py [hunk 2]
```diff
@@ -521,3 +523,32 @@ class A(object):
...
+
+    Example::
+
+         # Correct:
+         some_variable = 42
+
+         #Wrong:
+         переменная = 42
+
+    Note:
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/naming.py]: Please, remove `Note`

File: wemake_python_styleguide/violations/naming.py [hunk 3]
```diff
@@ -521,3 +523,32 @@ class A(object):
...
+
+    Solution:
+       Rename so that they contains only latin letters.
+
+    Example::
+
+         # Correct:
+         some_variable = 42
+
+         #Wrong:
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/naming.py]: This needs an extra space

File: wemake_python_styleguide/violations/naming.py [hunk 4]
```diff
@@ -521,3 +523,32 @@ class A(object):
...
+         переменная = 42
+
+    Note:
+        Returns Z116 as error code
+
+    """
+
+    #: Error message shown to the user.
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/naming.py]: Please, remove this comment

File: wemake_python_styleguide/violations/naming.py [hunk 5]
```diff
@@ -521,3 +523,32 @@ class A(object):
...
+    error_template = 'Found unicode name "{0}"'
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/naming.py]: `'Found unicode name: {0}'`

File: wemake_python_styleguide/visitors/ast/naming.py [hunk 6]
```diff
@@ -45,6 +48,8 @@
 class WrongNameVisitor(BaseNodeVisitor):
     """Performs checks based on variable names."""
 
+    unicode_check: ClassVar[Pattern] = re.compile(r'([a-zA-Z_0-9]+)')
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/naming.py]: This name needs to be protected: `_unicode_pattern`

File: wemake_python_styleguide/visitors/ast/naming.py [hunk 7]
```diff
@@ -59,6 +64,13 @@ def _check_name(self, node: ast.AST, name: str) -> None:
         if variables.is_variable_name_with_underscored_number(name):
             self.add_violation(UnderScoredNumberNameViolation())
 
+        """Check unicode symbols in name."""
+        matches = re.findall(self.unicode_check, name)
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/naming.py]: Why `findall`?

**Comment by geoc0ld** [comment]
@sobolevn something went wrong and now i can't update the my pull request. Can we close this request? After closing i will create a new pull request.

**Comment by sobolevn** [comment]
Sure!

**Pull request closed without merging**